plugins/extractors/maxcompute/client.go

```diff
@@ -0,0 +1,64 @@
+package maxcompute
+
+import (
+	"context"
+
+	"github.com/aliyun/aliyun-odps-go-sdk/odps"
+	"github.com/aliyun/aliyun-odps-go-sdk/odps/account"
+)
+
+type MCClient struct {
+	client  *odps.Odps
+	project *odps.Project
+}
+
+func NewMaxComputeClient(config Config) *MCClient {
+	aliAccount := account.NewAliyunAccount(config.AccessKey.ID, config.AccessKey.Secret)
+	client := odps.NewOdps(aliAccount, config.EndpointProject)
+	client.SetDefaultProjectName(config.ProjectName)
+
+	project := client.Project(config.ProjectName)
+
+	return &MCClient{
+		client:  client,
+		project: project,
+	}
+}
+
+func (c *MCClient) ListSchema(context.Context) (schemas []*odps.Schema, err error) {
+	err = c.project.Schemas().List(func(schema *odps.Schema, err2 error) {
+		if err2 != nil {
+			err = err2
+			return
+		}
+		schemas = append(schemas, schema)
+	})
+
+	return schemas, err
+}
+
+func (c *MCClient) ListTable(_ context.Context, schemaName string) (tables []*odps.Table, err error) {
+	t := odps.NewTables(c.client, c.project.Name(), schemaName)
+	t.List(
+		func(table *odps.Table, err2 error) {
+			if err2 != nil {
+				err = err2
```

Comment:
Should we stop if we get 1 error or log the error and continue for other tables?

Reply:
What I know that meteor will still process all tables that can be proceed, and skip for error tables. This is similar to [bigquery](https://github.com/goto/meteor/blob/main/plugins/extractors/bigtable/bigtable.go#L154-L162), so this is expected.

However, the one that need to change is [fetchTablesFromSchema](https://github.com/goto/meteor/pull/69/files#diff-ec7404c29a880d47a3a23923a2a6d6daf3c7f0380515ee82d758201d34bcb49cR149-R151) mas @Mayurjag . I believe the condition should be `if err != nil && len(tables) = 0 {`

CMIIW mas @haveiss

Reply:
however please add the log for error mas @Mayurjag

Reply:
addressed mas